hs_file_types/views.py

```diff
@@ -208,6 +210,89 @@ def remove_aggregation(request, resource_id, hs_file_type, file_type_id, **kwarg
     return JsonResponse(response_data, status=status.HTTP_200_OK)
 
 
+@login_required
+def delete_aggregation(request, resource_id, hs_file_type, file_type_id, **kwargs):
+    """Deletes all files associated with an aggregation and all the associated metadata.
+    """
+
+    response_data = {'status': 'error'}
+
+    res, _, _ = authorize(request, resource_id, needed_permission=ACTION_TO_AUTHORIZE.EDIT_RESOURCE)
+    if res.resource_type != "CompositeResource":
+        err_msg = "Aggregation type can be deleted only in composite resource."
+        response_data['message'] = err_msg
+        return JsonResponse(response_data, status=status.HTTP_400_BAD_REQUEST)
+
+    if hs_file_type not in FILE_TYPE_MAP:
+        err_msg = "Unsupported aggregation type. Supported aggregation types are: {}"
+        err_msg = err_msg.format(FILE_TYPE_MAP.keys())
+        response_data['message'] = err_msg
+        return JsonResponse(response_data, status=status.HTTP_400_BAD_REQUEST)
```

Comment:
It'd be potentially better performance-wise if lines 226 to 230 is moved before line 220-224 to skip access-control check for a file type-related bad request.

Reply:
Good suggestion. Fixed.